zeus/monitor/energy.py

```diff
@@ -274,12 +322,12 @@ def end_window(
         # Retrieve the start time and energy consumption of this window.
         try:
             measurement_state = self.measurement_states.pop(key)
-            start_time = measurement_state.time
-            gpu_start_energy = measurement_state.gpu_energy
-            cpu_start_energy = measurement_state.cpu_energy
-            dram_start_energy = measurement_state.dram_energy
         except KeyError:
             raise ValueError(f"Measurement window '{key}' does not exist") from None
+        start_time = measurement_state.time
+        gpu_start_energy = measurement_state.gpu_energy
+        cpu_start_energy = measurement_state.cpu_energy
+        dram_start_energy = measurement_state.dram_energy
```

Comment:
Move these to below the line that calls `end_time: float = time()`. The number of operations before taking the `end` timestamp must be minimized to make time measurement as accurate as possible,.